feature/navigation/drawer/dropdown/src/main/kotlin/net/thunderbird/feature/navigation/drawer/dropdown/ui/folder/FolderList.kt

```diff
@@ -41,16 +40,19 @@ internal fun FolderList(
             key = { it.displayFolder?.id ?: '0' },
         ) { folder ->
             val currentDisplayFolder = folder.displayFolder
-            if (currentDisplayFolder is DisplayAccountFolder) {
-                FolderListItem(
-                    displayFolder = currentDisplayFolder,
-                    treeFolder = folder,
-                    selected = currentDisplayFolder.folder == selectedFolder,
-                    showStarredCount = showStarredCount,
-                    onClick = onFolderClick,
-                    folderNameFormatter = folderNameFormatter,
-                )
-            }
+            FolderListItem(
+                displayFolder = currentDisplayFolder ?: run {
+                    throw NullPointerException(
+                        "Null DisplayFolder for folder ${folder.displayName}",
+                    )
+                },
```

Comment:
Could you please use `requireNotNull` instead of throwing `NullPointerException`?
```suggestion
                displayFolder = requireNotNull(currentDisplayFolder) {
                    "Null DisplayFolder for folder ${folder.displayName}"
                },
```